src/utils/isDomainValid.ts

```diff
@@ -0,0 +1,11 @@
+import ResolutionError, {ResolutionErrorCode} from "../errors/resolutionError";
+
+export function prepareDomain(domain: string): string {
+  const retVal: string = domain ? domain.trim().toLowerCase() : '';
+  if(!RegExp('^[.a-z\\d-]+$').test(retVal)){
```

Comment:
This validation regex can be super essential it is maybe a good idea to move it to config inside uns repo. cc @DeRain

Reply:
@bogdan I think it's premature to add this into shared config. We use this regexp only to throw correct error message if domain contains special characters.